investigations/Live Reduction/LiveReductionInvestigation.md

```diff
@@ -106,4 +106,67 @@ Number of OMP Threads|SANS2D (3s timeout)|WISH (60s timeout)
 4|264Hz|15Hz
 8|269Hz|14Hz
 12|278Hz|15Hz
-20|293Hz|15Hz
\ No newline at end of file
+20|293Hz|15Hz
+
+
+## Effects of Reduction Steps
+
+For this test, two technique areas were chosen to determine how performance degrades with an increasing number of algorithms. Each area, SANS and Powder Diffraction, were tested with a very basic and skeletal reduction script which contained around 3 algorithms and then with a more realistic script with >= 6 algorithm. Below are the scripts used for the tests.
+
+### SANS
+```python
+# SANSScript.py
+# SANS Reduction Basic
+Rebin(InputWorkspace="accum_events", OutputWorkspace="ws", Params="1,100,100006", PreserveEvents=False)
+ConvertUnits(InputWorkspace="ws", OutputWorkspace="ws", Target="Wavelength", AlignBins=True)
+Q1D(DetBankWorkspace="ws", OutputWorkspace="SANSReduced", OutputBinning="1,0.05,100")
+
+# SANS Reduction Realistic
+ConvertUnits(InputWorkspace="accum_events", OutputWorkspace="ws", Target="Wavelength")
+FindCenterOfMassPosition(InputWorkspace="ws", Output="CentreOfMass")
+com = mtd["CentreOfMass"]
+beam = com.column(1)
+MoveInstrumentComponent(Workspace="ws", ComponentName="rear-detector", X=beam[0], Y=beam[1], RelativePosition=True)
+MoveInstrumentComponent(Workspace="ws", ComponentName="front-detector", X=beam[0], Y=beam[1], RelativePosition=True)
+SANSSolidAngleCorrection(InputWorkspace="ws", OutputWorkspace="ws", DetectorTubes=True)
+SANSAbsoluteScale(InputWorkspace="ws", OutputWorkspace="ws", ScalingFactor=0.99)
+Rebin(InputWorkspace="ws", OutputWorkspace="ws", Params="1,100,100006", PreserveEvents=False)
+Q1D(DetBankWorkspace="ws", OutputWorkspace="SANSReduced", OutputBinning="1,0.05,100")
+```
+
+### Powder Diffraction
+```python
+# PowderDiffractionScript.py
+# Powder Diffraction Basic
+CompressEvents(InputWorkspace="accum_events", OutputWorkspace="ws")
+ConvertUnits(InputWorkspace="accum_events", OutputWorkspace="dspaced", Target="dSpacing")
+DiffractionFocussing(InputWorkspace="dspaced", OutputWorkspace="ws", GroupingFileName="/path/to/35922_h00_RW.cal", PreserveEvents=True)
+
+# Powder Diffraction Realistic
+CompressEvents(InputWorkspace="accum_events", OutputWorkspace="ws")
```

Comment:
If you call `CompressEvents` in the first step you are likely to see no big influence of the event rate on the time for the reduction steps. Can you try without?

Reply:
👍

Reply:
Removing `CompressEvents` didn't make a difference to the timings.

Reply:
Ok. I guess the effect would only become visible with more events.